core/src/main/scala/chisel3/Module.scala

```diff
@@ -138,6 +138,24 @@ object Module extends SourceInfoDoc {
     }
   }
 
+  /** Allowed values for the types of Module.reset */
+  object ResetType {
+
+    /** Allowed values for the types of Module.reset */
+    sealed trait Type
```

Comment:
```scala
   /** Allowed values for the types of Module.reset */
  sealed trait ResetType
  /** Allowed values for the types of Module.reset */
  object ResetType {

    /** The default reset type. This is Uninferred, unless it is the top Module, in which case it is Bool */
    case object Default extends ResetType
    ...
  }
```

I'm a little late here, but the more canonical way to write this would be having the object be the companion object to tye type.